### PR TITLE
MGMT-15902: Trigger reboots for node event when day2 node moves to done

### PR DESCRIPTION
**Pull request by ori-amizur**
This is done for kube api client only when spoke kubeconfig is available.
The implementation is done using debug command (as in oc command). The number of reboots are counted by 'last reboot' linux command.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [x] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md
/cc @filanov

### CODE REVIEW

File: internal/controller/controllers/debug_cmd.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		numWritten += n
+	}
+	if err != nil {
+		return "", err
+	}
+	return f.Name(), nil
+}
+
+func (d *debugCommands) runDebug(command string, args ...string) (out string, errOut string, err error) {
+	kcmdutil.BehaviorOnFatal(fatalHandler(d.log))
```
Comment (tsorya) [inline internal/controller/controllers/debug_cmd.go:81]: can you explain? cause it is running in routine?

File: internal/controller/controllers/agent_controller.go [hunk 2]
```diff
@@ -484,6 +487,31 @@ func (r *AgentReconciler) spokeKubeClient(ctx context.Context, clusterRef *aiv1b
...
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to get spoke secret for cluster %s/%s", clusterRef.Namespace, clusterRef.Name)
+		return err
+	}
+	kubeconfig, err := spoke_k8s_client.KubeconfigFromSecret(secret)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to get kubeconfig from secret for cluster %s/%s", clusterRef.Namespace, clusterRef.Name)
+		return err
+	}
+	cmd := NewDebugCommands(r.Log, kubeconfig)
```
Comment (tsorya) [inline internal/controller/controllers/agent_controller.go:504]: why not to provide client?

File: internal/controller/controllers/debug_cmd.go (hunk 1 above)
Comment (ori-amizur) [reply internal/controller/controllers/debug_cmd.go:81]: The default fatal behavior is to exit (which we don't want).  So I replaced with panic (and recover).
The expectation is that no code will be executed after fatal handler is called.

File: internal/controller/controllers/agent_controller.go (hunk 2 above)
Comment (ori-amizur) [reply internal/controller/controllers/agent_controller.go:504]: There was a request to use this as the image is already too large

Comment (tsorya) [reply internal/controller/controllers/agent_controller.go:504]: Now i understand, you actually run it as oc command

File: internal/controller/controllers/debug_cmd.go (hunk 1 above)
Comment (ori-amizur) [reply internal/controller/controllers/debug_cmd.go]: Changed

File: internal/controller/controllers/agent_controller.go [hunk 3]
```diff
@@ -484,6 +488,31 @@ func (r *AgentReconciler) spokeKubeClient(ctx context.Context, clusterRef *aiv1b
 	return r.SpokeK8sClientFactory.CreateFromSecret(secret)
 }
 
+func (r *AgentReconciler) notifyNumberOfReboots(ctx context.Context, h *models.Host, node *corev1.Node, clusterRef *aiv1beta1.ClusterReference) error {
+	if !isNodeReady(node) {
```
Comment (tsorya) [inline internal/controller/controllers/agent_controller.go:492]: Ready is enough to run debug pod?

File: internal/oc/debug.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	}, args...)
+	out, errOut, exitCode := d.exec.Execute("oc", execArgs...)
+	if exitCode != 0 {
+		return "", "", errors.Errorf("oc debug failed to execute with code %d: %s", exitCode, errOut)
+	}
+	return out, errOut, nil
+}
+
+func (d *debug) RebootsForNode(nodeName string) (int, error) {
+	out, _, err := d.runDebug("node/"+nodeName,
```
Comment (tsorya) [inline internal/oc/debug.go:76]: I think this command can stuck for a very long time and we should have timeout for it

Comment (ori-amizur) [reply internal/oc/debug.go:76]: OK

**Pull request merged**